### PR TITLE
fix-1551: mostly traits docs.

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
I don't know how to doxygen. I will ask you to do it after, OK?

### CODE REVIEW

File: doc/tutorial/frequency-scaling.hpp [hunk 1]
```diff
@@ -23,9 +23,15 @@ you can pass `eve::algo::allow_frequency_scaling` trait.
 There are also a typedefs `nofs_wide`, `nofs_logical` where `nofs` stands for
 "no frequency scaling".
 
+If you would like the default register size to be 64 bytes on AVX512, for example
+because you are on AMD where this problem does not exist, or you just have a domain
+where that makes sense - you can define EVE_AVX512_DEFAULT_64_BYTES on the command line.
```
Comment (jfalcou) [inline doc/tutorial/frequency-scaling.hpp:28]: in your compilation options.
People may not use command line to compile alas.

File: doc/tutorial/interacting-with-native-code.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#error This file is for documentation only - DO NOT INCLUDE
+/**
+
+@page inter-with-native Interaction with native code.
+
+In EVE we recognise that we cannot possibly cover all the
+individual things you might want to do and it's necessery for
+the users to be able to write intrinsics.
+
+That's why eve::wide/eve::logical for native sizes are implicitly
```
Comment (jfalcou) [inline doc/tutorial/interacting-with-native-code.hpp:10]: eve::wide and eve::logical

the / prevent auto-link to work.

File: doc/tutorial/interacting-with-native-code.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+convertible to and from the intrinsic type you'd expect.
+
+x86: __m128/256/512 for floats and double, __m128i/256i/512i for ints
```
Comment (jfalcou) [inline doc/tutorial/interacting-with-native-code.hpp:13]: Make a real bullet point
```
  * x86: ....
  * neon: ...
  * ...
```

File: doc/tutorial/interacting-with-native-code.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+@snippet tutorial/interacting_with_native_code.cpp interating_with_native_algo
+
+# What about non-native sizes?
+
+If the cardinal is less than can be natively represented, it's internally
+still represented as a full register. For example wide<int, eve::fixed<2>>
+on x86 is still __m128i. The data is at the first 2 elements, the rest can
+be garbage.
+
+If the cradinal is bigger than natively represenatble, you can use slice to
```
Comment (jfalcou) [inline doc/tutorial/interacting-with-native-code.hpp:30]: If the **cardinal** is larger than **the one natively representable**

File: include/eve/module/algo/algo/traits.hpp [hunk 5]
```diff
@@ -20,6 +20,31 @@
 
 namespace eve::algo
 {
+  //================================================================================================
```
Comment (jfalcou) [inline include/eve/module/algo/algo/traits.hpp:23]: You can save on line by having a single
```
//================================================================================================
//! @addtogroup algo_traits
//! @{
//================================================================================================
```

on top and a single 

```
//================================================================================================
//! @}
//================================================================================================
```

at the end.

**Review by jfalcou — CHANGES_REQUESTED** [review]
LGTM when the smalle ditorial changes are done

File: include/eve/module/algo/algo/traits.hpp (hunk 5 above)
Comment (DenisYaroshevskiy) [reply include/eve/module/algo/algo/traits.hpp:23]: How about you do this docker things so that unnecessary things are not listed?

**Review by jfalcou — APPROVED** [review]

**Pull request merged**